core/src/test/scala/com/salesforce/op/stages/impl/feature/AttributeAsserts.scala

```diff
@@ -42,8 +43,11 @@ trait AttributeAsserts {
    * @param schema
    * @param expectedNominal Expected array of booleans. True if the field is nominal, false if not.
    */
-  final def assertNominal(schema: StructField, expectedNominal: Array[Boolean]): Assertion = {
+  final def assertNominal(schema: StructField, expectedNominal: Array[Boolean], output: Array[OPVector]): Assertion = {
     val attributes = AttributeGroup.fromStructField(schema).attributes
+    output.foreach{
+      x => x.value.toArray.zip(expectedNominal).foreach{ case (v , n) => if (n) (v == 0.0 || v == 1.0) shouldBe true }
```

Comment:
should we also add a check when `n == false`? is so, then better:
```scala
for {
    x <- output
    (value, nominal) <- x.value.toArray.zip(expectedNominal)
} if (nominal) (v == 0.0 || v == 1.0) shouldBe true else ???
```

Reply:
there are no constraints on if n is false

Reply:
so it can be NaN, -/+Inf as well?